logging/impl.go

```diff
@@ -474,3 +509,325 @@ func getCaller() zapcore.EntryCaller {
 
 	return entryCaller
 }
+
+func (imp *implWith) Desugar() *zap.Logger {
+	return imp.AsZap().Desugar()
+}
+
+func (imp *implWith) SetLevel(level Level) {
+	imp.logger.SetLevel(level)
+}
+
+func (imp *implWith) GetLevel() Level {
+	return imp.logger.GetLevel()
+}
+
+func (imp *implWith) Level() zapcore.Level {
+	return imp.GetLevel().AsZap()
+}
+
+func (imp *implWith) Sublogger(subname string) Logger {
+	return imp.logger.Sublogger(subname)
+}
+
+func (imp *implWith) Named(name string) *zap.SugaredLogger {
+	return imp.AsZap().Named(name)
+}
+
+func (imp *implWith) Sync() error {
+	return imp.logger.Sync()
+}
+
+func (imp *implWith) WithOptions(opts ...zap.Option) *zap.SugaredLogger {
+	return imp.AsZap().WithOptions(opts...)
+}
+
+func (imp *implWith) AddAppender(appender Appender) {
+	imp.logger.AddAppender(appender)
+}
+
+func (imp *implWith) AsZap() *zap.SugaredLogger {
+	return imp.logger.AsZap()
+}
+
+func (imp *implWith) Write(entry *LogEntry) {
+	imp.logger.Write(entry)
+}
+func (imp *implWith) WithFields(args ...interface{}) Logger {
+	return imp.logger.WithFields(args)
+}
+
+func (imp *implWith) Debug(args ...interface{}) {
+	imp.logger.testHelper()
+	if imp.logger.shouldLog(DEBUG) {
+		entry := imp.logger.format(DEBUG, emptyTraceKey, args...)
+		entry.fields = append(entry.fields, imp.logFields...)
+		imp.Write(entry)
+	}
+}
```

Comment:
Calling `imp.logger.Debugw(...)` here with `imp.logFields` also works, but this makes it so that the file name / line number don't properly get logged, which is fixed by just copying and slightly modifying implementation of `impl` functions.

Reply:
> this makes it so that the file name / line number don't properly get logged

This is _probably_ somehow avoidable, but I'm personally fine with the copying. The risk we do run is that we fix some issue in `impl` and forget to fix it in `implWith`.